app/scripts/controllers/bridge/bridge-controller.ts

```diff
@@ -49,4 +60,28 @@ export default class BridgeController extends BaseController<
       _state.bridgeState = { ...bridgeState, bridgeFeatureFlags };
     });
   };
+
+  selectDestNetwork = async (chainId: Hex) => {
+    await this.#setTopAssets(chainId, 'destTopAssets');
+    await this.#setTokens(chainId, 'destTokens');
+  };
+
+  #setTopAssets = async (
+    chainId: Hex,
+    stateKey: 'srcTopAssets' | 'destTopAssets',
+  ) => {
+    const { bridgeState } = this.state;
+    const topAssets = await fetchTopAssetsList(chainId);
+    this.update((_state) => {
+      _state.bridgeState = { ...bridgeState, [stateKey]: topAssets };
+    });
```

Comment:
I wonder if it would make sense to store top assets and bridge-able tokens by chain id so (1) we won't need to wait for a refetch if the user switches back and forth between a few networks and (2) we could pre-fetch all the values for all chains before the user makes a selection so the user won't have to wait for the api call to return before making a selection. Wdyt?

Reply:
+1 to this

Reply:
Currently there's only a loading delay the first time a network is selected. The token data and top assets responses are cached in the browser so we don't re-fetch when the user switches back and forth. Some token lists are large (ethereum has 3000+ for example) so I chose to only load the src and dest lists to minimize the amount of unused state data and avoid bloating state. Persisting the data across sessions is also easier to manage with browser caching

I like the idea of prefetching, but it might not completely remove the delay when initially loading a network since we need to wait for data coming from other controllers, such as token prices, balances, gas, etc. Definitely open to trying it out though. For now, can I add this to the list of post-MVP improvements we can investigate later on?

Reply:
I'm good with pushing this off for a post MVP optimization